CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Joel Winarske
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+cmake_minimum_required(VERSION 3.10.2)
+
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build, options are: Debug, Release, or MinSizeRel." FORCE)
+    message(STATUS "CMAKE_BUILD_TYPE not set, defaulting to Release.")
+endif()
+
+project(flutter-pi LANGUAGES C)
+
+message(STATUS "Generator .............. ${CMAKE_GENERATOR}")
+message(STATUS "Build Type ............. ${CMAKE_BUILD_TYPE}")
+
+if(NOT FLUTTER_ENGINE_LIBRARY)
+  set(FLUTTER_ENGINE_SHA af51afceb8886cc11e25047523c4e0c7e1f5d408)
```

Comment:
Four things:

- Allow the user to specify a flutter channel. If the user specifies a flutter channel for building, use the latest engine SHA for that channel for retrieving the artifacts.
- Allow the user to specify the engine SHA if no channel was given. Breaking changes (in the header for example) do happen.
- Generally, inform the user if the build script makes a decision _for_ them. (Like, choosing an engine SHA)
- Have the script make a more reasonable default choice of the engine SHA. (Latest stable for example)

The latest engine SHA for a given flutter channel can be retrieved from
```
https://github.com/flutter/flutter/blob/CHANNEL/bin/internal/engine.version
```

Reply:
Updated to allow overriding SHA1.

I agree with your points.  CMake is a project generator opposed to general script.  If you're looking for that level of interaction then it would be best to wrap it with a bash or python script.  The CMake variables represent the build API, and keeps it versatile.

Reply:
I'll look into adding channel selection.  Default to stable or beta?

Reply:
I added channel selection.  It defaults to `stable` if none is selected, otherwise you switch channels like this:
    `cmake .. -DCHANNEL=stable`
    `cmake .. -DCHANNEL=beta`
    `cmake .. -DCHANNEL=dev`
    `cmake .. -DCHANNEL=master`